module/zfs/dbuf.c

```diff
@@ -619,7 +720,24 @@ dbuf_evict_notify(void)
 	}
 }
 
+void
+dbuf_flush_impl(void)
+{
+	unsigned int retries = 0;
+	int64_t prev_size = -1;
+	int64_t size = refcount_count(&dbuf_cache_size);
+
+	while (size > 0 && retries < 10) {
+		dbuf_evict_one();
 
+		size = refcount_count(&dbuf_cache_size);
+		if (size < prev_size)
+			retries = 0;
+		else
+			retries++;
+		prev_size = size;
+	}
+}
```

Comment:
We shouldn't add an entirely new module option for this, particularly without a clear use case in mind.  We have more than we need already.  I could potentially see an argument for adding a `dbuf_cache` token to `arc_flush`.